spec/lib/excon/zipkin-tracer_spec.rb

```diff
@@ -51,6 +51,34 @@ def process(body, url, headers = {})
     end
   end
 
+  it 'has a trace with correct duration' do
+    Trace.tracer = Trace::NullTracer.new
+    ::Trace.sample_rate = 1
+    trace_id = ::Trace::TraceId.new(1, 2, 3, true, ::Trace::Flags::DEBUG)
+    url = 'https://www.example.com'
+    allow(::Trace).to receive(:default_endpoint)
+      .and_return(::Trace::Endpoint.new('127.0.0.1', '80', 'example.com'))
+    stub_request(:post, url)
+      .to_return(body: lambda { |request| sleep 1; '' })
```

Comment:
clever. I think eventually the orchestration around this could be refactored. The key fact lost here is how we end up with 1_000_000. I think it is because of the sleep. Might be worth making the argument to sleep a variable like duration_seconds, and multiply that in the  later assertion will make more sense.